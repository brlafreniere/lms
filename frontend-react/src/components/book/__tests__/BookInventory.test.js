import React from "react";
import Axios from "axios";
import renderer from "react-test-renderer";

import BookInventoryForm from "../BookInventory";
import Book from "../../../modules/book";
import { MemoryRouter } from "react-router-dom";

test("<BookDetail /> snapshot test", () => {
    const book_inventories = [
        {
            id: 1,
            title: "Foobar",
            cover_image_file_name: "foo.jpg",
            branch: {
                name: "Willows Peak"
            }
        }
    ]

    Book.fetchInventories = jest.fn()
    Book.fetchInventories.mockResolvedValue(book_inventories);

    const component = renderer.create(
        <MemoryRouter>
            <BookInventoryForm book_id={4} />
        </MemoryRouter>
    )

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});