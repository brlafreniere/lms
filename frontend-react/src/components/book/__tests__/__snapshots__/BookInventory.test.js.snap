// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookDetail /> snapshot test 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <button
      className="btn btn-primary mb-3"
    >
      Add Inventory to Branch
    </button>
    <table
      className="table table-bordered table-hover"
    >
      <colgroup>
        <col
          span="1"
          style={
            Object {
              "width": "75%",
            }
          }
        />
        <col
          span="1"
          style={
            Object {
              "width": "25%",
            }
          }
        />
      </colgroup>
      <thead
        className="thead-dark"
      >
        <tr>
          <th>
            Branch
          </th>
          <th>
            Copies
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </form>
</div>
`;
